### ISSUE TITLE
Exact learning rate schedule

### ISSUE THREAD

**Issue by ducha-aiki** [vbalnt/tfeat#6]
Paper says:

>. For the optimization the Stochastic Gradient Descend [3] is used, and the training is done in batches
of 128 items, with a learning rate of 0.1 which is temporally annealed, momentum of 0.9
and weight decay of 10−6 . We also reduce the learning rate every epoch

Could you please clarify, which exactly factor is used for reducing learning rate schedule? And which epoch snapshot is the one used for getting results in paper?

**Comment by ducha-aiki** [comment]
@edgarriba Could you please answer this one? Results I have got in Liberty are much worse that weights  which are online.

**Comment by vbalnt** [comment]
Hi @ducha-aiki, 

I am not sure about the `pytorch` version, but for the training schedule in `torch` that is where the results in the paper are from:

Training was done with *11M* randomly sampled triplets. Note that in a first version of the paper, we were training with much less around *1M*), but since we have found that increasing the number of training triplets increases the performance. 


SGD parameters for the `torch` code
```
learningRate = 0.1,
weightDecay = 1e-4,
momentum = 0.9,
learningRateDecay = 1e-6
```

For details on the learning rate annealing you can check the code from `torch`
https://github.com/torch/optim/blob/master/sgd.lua#L71

I think that the `pytorch` code uses the same optim library, so if you put in this configuration, with *11M* training triplets, might lead to similar results. 

You can train until convergence which typically is around 150-200 epochs. The results are at the 200th epoch snapshot.

**Comment by ducha-aiki** [comment]
@vbalnt thanks, I will try this setup.

**Comment by ducha-aiki** [comment]
@vbalnt 
At epoch 100 (train on Liberty, test on Notredame) FPR95 ~ 0.0468 and doesn`t really change for ten epochs. I will continue, but it is unlikely to converge to 3.12 as in paper...

**Comment by edgarriba** [comment]
tensorflow or pytorch?

**Comment by ducha-aiki** [comment]
@edgarriba pytorch. 
But I have found, that shuffling is off by my mistake, when I have switched test and train sets (in your example, train is Notredame). I am not sure if this will help for 1% acc, but will see.
